File: kotlin/functions/string_manipulation.kt
fun main() {
    val text = "  Hello, World!  "

    println("Original: '$text'")
    println("Upper: '${text.uppercase()}'")
    println("Lower: '${text.lowercase()}'")
    println("Trim: '${text.trim()}'")
    println("Replace: '${text.replace("World", "Kotlin")}'")
    println("Split: ${text.trim().split(", ")}")
    println("Contains: ${text.contains("World")}")
    println("StartsWith: ${text.trim().startsWith("Hello")}")
    println("EndsWith: ${text.trim().endsWith("!")}")
    println("Length: ${text.length}")
    println("Substring: '${text.trim().substring(0, 5)}'")
    println("Repeat: '${"-".repeat(20)}'")
    println("Reversed: '${text.trim().reversed()}'")
}

File: csharp/functions/StringManipulation.cs
using System;

class StringManipulation
{
    static void Main()
    {
        string text = "  Hello, World!  ";

        Console.WriteLine($"Original: '{text}'");
        Console.WriteLine($"Upper: '{text.ToUpper()}'");
        Console.WriteLine($"Lower: '{text.ToLower()}'");
        Console.WriteLine($"Trim: '{text.Trim()}'");
        Console.WriteLine($"Replace: '{text.Replace("World", "C#")}'");
        Console.WriteLine($"Split: {string.Join(", ", text.Trim().Split(", "))}");
        Console.WriteLine($"Contains: {text.Contains("World")}");
        Console.WriteLine($"StartsWith: {text.Trim().StartsWith("Hello")}");
        Console.WriteLine($"EndsWith: {text.Trim().EndsWith("!")}");
        Console.WriteLine($"Length: {text.Length}");
        Console.WriteLine($"Substring: '{text.Trim().Substring(0, 5)}'");
    }
}
